src/Farfetch.LoadShedding/Tasks/ConcurrentCounter.cs

```diff
@@ -64,10 +64,7 @@ public int Decrement()
         {
             lock (this._locker)
             {
-                if (this._count > 0)
-                {
-                    this._count--;
-                }
+                this._count--;
```

Comment:
Is possible to have the `_count` with 0, and when entering the method, `_count` passes to negative values?

Reply:
this validation was hidding an error, it is not possible to have _count < 0 since each queue task should be increased and decreased in the final

Reply:
BTW I've reverted the change and changed the unit test to avoid the error to occur again